crates/ruff/src/rules/pyupgrade/rules/unnecessary_coding_comment.rs

```diff
@@ -67,7 +67,7 @@ pub(crate) fn unnecessary_coding_comment(
             let mut diagnostic = Diagnostic::new(UTF8EncodingDeclaration, *range);
             if settings.rules.should_fix(diagnostic.kind.rule()) {
                 diagnostic.set_fix(Fix::automatic(Edit::deletion(
-                    range.start(),
+                    locator.line_start(range.start()),
                     locator.full_line_end(range.end()),
```

Comment:
What happens if the coding comment is preceded by content, like:

```python
print(x) # coding=utf8
print("Hello world")
```

Reply:
That shouldn't match our regex, but I can add test coverage

Reply:
Except it does get removed.. humph will investigate

Reply:
The regular expression is only run over the comment (the range is that of the comment).

Reply:
Yeah I just realized that — this was less clear when the variable was named `line`, pretty obvious now

Reply:
[PEP 263](https://peps.python.org/pep-0263/) doesn't let you declare these as comments with other content on the line so I've updated the regex to apply to the full line.